src/execution/helper.rs

```diff
@@ -234,7 +234,32 @@ trait GenCallIter {
 impl GenCallIter for TransactionExecutionInfo {
     fn gen_call_iterator(&self) -> IntoIter<CallInfo> {
         let mut call_infos = vec![];
-        for call_info in self.non_optional_call_infos() {
+
+        // Determine if we are treating a DEPLOY_ACCOUNT tx. For deployments we need
+        // to order call infos differently, __validate_deploy__ is called after the constructor.
+        // See https://docs.starknet.io/architecture-and-concepts/accounts/account-functions/#overview
+        // for more details.
+        let is_deploy = match &self.execute_call_info {
+            Some(call_info) => matches!(call_info.call.entry_point_type, EntryPointType::Constructor),
+            None => false,
+        };
+
+        let call_info_iter = match is_deploy {
```

Comment:
`match` on boolean is a bit unusual, you can also write as:

```rust
let foo = if condition { val1 } else { val2 }
```

This is purely cosmetic and not a big deal though :)

Reply:
Good point!